frontend/src/router/guards.js

```diff
@@ -36,19 +36,42 @@ export function createGlobalBeforeGuards () {
   const shootStore = useShootStore()
   const terminalStore = useTerminalStore()
 
+  let sessionTimeoutId
+
   function ensureUserAuthenticatedForNonPublicRoutes () {
-    return (to) => {
-      const { meta = {}, path } = to
+    return to => {
+      const {
+        meta = {},
+        fullPath: redirectPath,
+      } = to
+
+      if (meta.public) {
+        return true
+      }
+
       authnStore.$reset()
-      if (!meta.public && authnStore.isExpired()) {
-        logger.info('User not found or session has expired --> Redirecting to login page')
-        const query = path !== '/'
-          ? { redirectPath: path }
-          : undefined
-        return {
-          name: 'Login',
-          query,
-        }
+
+      const expiresIn = authnStore.userExpiresAt - Date.now()
```

Comment:
shouldn't we rename `userExpiresAt` to `userSessionExpiresAt` or `sessionExpiresAt` to make it clearer?

Reply:
renamed to `sessionExpiresAt`